### PR TITLE
chore: add test cases

### PR DESCRIPTION
**Pull request by mathur07**

### CODE REVIEW

File: tests/api/test_api_v2.py [hunk 1]
```diff
@@ -124,6 +124,86 @@ def test_get_component_analyses_bad_request(self):
...
+        accept_json = [('Content-Type', 'application/json;')]
+        response = self.client.post(
+            api_route_for('/vulnerability-analysis'), data=json.dumps(payload), headers=accept_json)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json, self.recommendation_data)
+
+    def test_get_vulnerability_analysis_bad_request(self):
+        """VA POST: Bad Request."""
+        payload = {
+            "ecosys": 'pypi',
```
Comment (rafiu007) [inline tests/api/test_api_v2.py:173]: This should be "ecosystem" .
and For python this should not be the response.
currently only golang is not supported

**Review by rafiu007 — APPROVED** [review]
LGTM. Suggested a minior change.
Wait for review from @yzainee

**Review by yzainee-zz — APPROVED** [review]
LGTM. I would like to see a thing added though. As the fixed_in field is a list, try adding 2-3 values in the graph mock response and see if you get all the 3 in the response you send back from the api.

**Review by rafiu007 — APPROVED** [review]
LGTM

**Pull request merged**